code/processes/wdb.q

```diff
@@ -313,7 +313,15 @@ merge:{[dir;pt;tableinfo;mergelimits;hdbsettings]
 	      curr[0]:curr[0],select from get segment;
 	      curr[1]:curr[1],segment;		
       	$[islast or mergemaxrows < count curr[0];
-	        [.lg.o[`merge;"upserting ",(string count curr[0])," rows to ",string dest];
+	        
+                [pattrtest:@[{@[x;y;`p#]}[curr[0];];`sym;{x}];
+                if[10h=type pattrtest;
+                  /p attribute could not be applied, data must be re-sorted by subpartition col (sym):
+                  curr[0]: xasc[.merge.getextrapartitiontype[tablename];curr[0]];
```

Comment:
Would be good to have a "re-sorting data" type log line before it does the sort as well